server/src/session.ts

```diff
@@ -1164,41 +1167,47 @@ export class Session {
     this.disableLanguageServiceForProject(project, 'ngcc is running');
     const configFilePath = project.getConfigFilePath();
 
-    this.connection.sendProgress(NgccProgressType, NgccProgressToken, {
-      done: false,
-      configFilePath,
-      message: `Running ngcc for ${configFilePath}`,
-    });
+    const progressReporter = await this.connection.window.createWorkDoneProgress();
 
-    let success = false;
+    progressReporter.begin('Angular', 0, `Running ngcc for ${configFilePath}`);
 
     try {
       this.projectNgccQueue.push({project, done: false});
       await resolveAndRunNgcc(configFilePath, {
         report: (msg: string) => {
-          this.connection.sendProgress(NgccProgressType, NgccProgressToken, {
-            done: false,
-            configFilePath,
-            message: msg,
-          });
+          progressReporter.report(50, msg);
```

Comment:
I'm not a huge fan of the percentage. The first message is definitely not an indicator of 50% and there's really no way to tell an accurate percent. Can we use `undefined` in the begin and omit the number parameter here?

This isn't a huge deal if you don't think that makes sense.